library/src/CMakeLists.txt

```diff
@@ -74,6 +74,12 @@ prepend_path( ".." rocfft_headers_public relative_rocfft_headers_public )
 
 add_library( rocfft ${rocfft_source} ${relative_rocfft_headers_public} )
 add_library( roc::rocfft ALIAS rocfft )
+
+# RUNPATH is set only when ROCM_RPATH is defined in the ENV
+if( DEFINED ENV{ROCM_RPATH} )
+  set ( CMAKE_SHARED_LINKER_FLAGS " -Wl,--enable-new-dtags -Wl,--rpath,$ENV{ROCM_RPATH}" )
```

Comment:
If this works, I believe we need this for rocfft-device build as well.

Reply:
This works, we have made a complete ROCm 2.10.1 build based on these changes. QA tested all the cases and approved the build also.

Reply:
The `CMAKE_SHARED_LINKER_FLAGS` variable should be changed by the toolchain and not in the CMakeLists.txt files.

Reply:
I will move it toolchain.